src/iiif/ingress_zip_consumer.py

```diff
@@ -0,0 +1,79 @@
+import json
+import logging
+import os
+import shutil
+
+from django.conf import settings
+from ingress.consumer.base import BaseConsumer
+
+from iiif import tools
+from main import settings
+
+logger = logging.getLogger(__name__)
+
+
+class ZipConsumer(BaseConsumer):
+    collection_name = settings.ZIP_COLLECTION_NAME
+
+    """
+    Whether or not to immediately remove messages once consumption succeeds.
+    If set to False, message.consume_succeeded_at will be set.
+    """
+    remove_message_on_consumed = False
+
+    """
+    Whether or not to set Message.consume_started_at immediately once consumption starts
+    """
+    set_consume_started_at = True
+
+    def consume_raw_data(self, raw_data):
+        record = json.loads(raw_data)
+
+        # Create a tmp folder to store downloaded source files
+        zipjob_uuid, tmp_folder_path = tools.create_tmp_folder()
+
+        # Init contents of txt info file which is send along in the zip
+        info_txt_contents = "The following files were requested:\n"
+        for i, (iiif_url, info) in enumerate(record['urls'].items()):
+            info_txt_contents += f"{iiif_url}: "
+            try:
+                file_response, file_url = tools.get_file(
+                    record['request_meta'], info['url_info'], iiif_url, info['metadata'])
+                tools.handle_file_response_errors(file_response, file_url)
+            except tools.ImmediateHttpResponse as e:
+                # Write small file
```

Comment:
I don't understand this comment

Reply:
No idea what that was supposed to mean..